http-agent/src/main/java/org/motechproject/http/agent/listener/HttpClientEventListener.java

```diff
@@ -200,6 +200,13 @@ public void handle(MotechEvent motechEvent) {
         try {
             retValue = r.call();
         } catch (Exception e) { // Http request failed for all retries
+            if (e instanceof HttpException && !(e.getCause().toString().contains("I/O error"))) {
```

Comment:
Verifying whether a string representation of the cause exception contains something is wrong